pkg/ttnpb/field_mask_validation.go

```diff
@@ -0,0 +1,577 @@
+		"pending_session.keys.nwk_s_enc_key.kek_label",
+		"pending_session.keys.nwk_s_enc_key.key",
+		"pending_session.keys.s_nwk_s_int_key",
+		"pending_session.keys.s_nwk_s_int_key.kek_label",
+		"pending_session.keys.s_nwk_s_int_key.key",
+		"pending_session.keys.session_key_id",
+		"pending_session.last_conf_f_cnt_down",
+		"pending_session.last_f_cnt_up",
+		"pending_session.last_n_f_cnt_down",
+		"pending_session.started_at",
+		"power_state",
+		"queued_application_downlinks",
+		"recent_adr_uplinks",
+		"recent_downlinks",
+		"recent_uplinks",
+		"session",
+		"session.dev_addr",
+		"session.keys",
+		"session.keys.f_nwk_s_int_key",
+		"session.keys.f_nwk_s_int_key.kek_label",
+		"session.keys.f_nwk_s_int_key.key",
+		"session.keys.nwk_s_enc_key",
+		"session.keys.nwk_s_enc_key.kek_label",
+		"session.keys.nwk_s_enc_key.key",
+		"session.keys.s_nwk_s_int_key",
+		"session.keys.s_nwk_s_int_key.kek_label",
+		"session.keys.s_nwk_s_int_key.key",
+		"session.keys.session_key_id",
+		"session.last_conf_f_cnt_down",
+		"session.last_f_cnt_up",
+		"session.last_n_f_cnt_down",
+		"session.started_at",
+		"supports_class_b",
+		"supports_class_c",
+		"supports_join",
+		"updated_at",
+		"version_ids",
+		"version_ids.brand_id",
+		"version_ids.firmware_version",
+		"version_ids.hardware_version",
+		"version_ids.model_id",
+	},
+	"/ttn.lorawan.v3.NsEndDeviceRegistry/Set": {
+		"battery_percentage",
+		"downlink_margin",
+		"frequency_plan_id",
+		"ids",
+		"ids.application_ids",
+		"ids.application_ids.application_id",
+		"ids.dev_addr",
+		"ids.dev_eui",
+		"ids.device_id",
+		"ids.join_eui",
+		"lorawan_phy_version",
+		"lorawan_version",
+		"mac_settings",
+		"mac_settings.adr_margin",
+		"mac_settings.class_b_timeout",
+		"mac_settings.class_c_timeout",
+		"mac_settings.desired_rx1_data_rate_offset",
+		"mac_settings.desired_rx1_delay",
+		"mac_settings.desired_rx1_delay.value",
+		"mac_settings.desired_rx2_data_rate_index",
+		"mac_settings.desired_rx2_data_rate_index.value",
+		"mac_settings.desired_rx2_frequency",
+		"mac_settings.factory_preset_frequencies",
+		"mac_settings.max_duty_cycle",
+		"mac_settings.max_duty_cycle.value",
+		"mac_settings.ping_slot_data_rate_index",
+		"mac_settings.ping_slot_data_rate_index.value",
+		"mac_settings.ping_slot_frequency",
+		"mac_settings.ping_slot_periodicity",
+		"mac_settings.ping_slot_periodicity.value",
+		"mac_settings.resets_f_cnt",
+		"mac_settings.rx1_data_rate_offset",
+		"mac_settings.rx1_delay",
+		"mac_settings.rx1_delay.value",
+		"mac_settings.rx2_data_rate_index",
+		"mac_settings.rx2_data_rate_index.value",
+		"mac_settings.rx2_frequency",
+		"mac_settings.status_count_periodicity",
+		"mac_settings.status_time_periodicity",
+		"mac_settings.supports_32_bit_f_cnt",
+		"mac_settings.use_adr",
+		"mac_state.device_class",
+		"mac_state.lorawan_version",
+		"mac_state.ping_slot_periodicity",
+		"max_frequency",
+		"min_frequency",
+		"power_state",
+		"session.dev_addr",
```

Comment:
@rvolosatovs 

1. Not `session` as a whole?
2. Why not `pending_session` for migration?

If there are good arguments then please apply this to AS as well; we need to be consistent.

Reply:
For now writing internal NS fields may not be fully supported, I designed it with idea that only NS writes it's own internal fields - being able to change internal device state in NS was not defined as a feature, hence I did not work towards supporting it.
I need to make sure it does not break anything in NS first, but that's out of scope of this PR. Once I'm sure it will work, I will add it here